Multitouch Support/Native/VoodooI2CNativeEngine.cpp

```diff
@@ -137,14 +139,32 @@ bool VoodooI2CNativeEngine::isForceClickEnabled() {
         lastForceClickPropertyUpdateTime = now_abs;
         return lastIsForceClickEnabled;
     }
-
+    
     // Blocking reading here takes about 10 microseconds
     OSDictionary* dict = OSDynamicCast(OSDictionary, this->getProperty("MultitouchPreferences", gIOServicePlane, kIORegistryIterateRecursively));
     if (!dict) {
         return lastIsForceClickEnabled;
     }
     if (OSCollectionIterator* i = OSCollectionIterator::withCollection(dict)) {
         while (OSSymbol* key = OSDynamicCast(OSSymbol, i->getNextObject())) {
+            // If tap to click (property "Clicking") is disabled,
+            // then we should forbid force clicking,
+            // because it makes impossible to perform a non-force click
+            if (key->isEqualTo("Clicking")) {
+                OSBoolean *boolValue;
+                OSNumber *numValue;
+                if ((boolValue = OSDynamicCast(OSBoolean, dict->getObject(key))) != NULL) {
```

Comment:
Stylistic: This is a bit hard to read, can we perhaps please refactor the conidition to the following:

```
boolValue = OSDynamicCast(OSBoolean, dict->getObject(key);
if (boolValue != NULL && !boolValue->getValue()) { ....

```

This reduces the extra if statement as well.

Reply:
No, it doesn't reduce the extra `if` statement. The if should be nested, because `else` should be executed only if cast failed (it's not a boolean), but not if `getValue()` returned true.